lib/gitOpsApi.js

```diff
@@ -115,25 +107,24 @@ module.exports = {
 	 * {isError: true, message: [error-message]};
 	 */
 	deleteBranch: async (octokit, options) => {
-		return new Promise((resolve, reject) => {
-			return octokit.request("DELETE /repos/{owner}/{repo}/git/refs/{ref}", {
+		try {
+			const response = await octokit.request("DELETE /repos/{owner}/{repo}/git/refs/{ref}", {
 				owner: options.repoOwner,
 				repo: options.repoName,
 				ref: "heads/" + options.branchName
-			}).then(() => {
-				resolve(options.branchName + " has been deleted successfully.");
-			}).catch((e) => {
-				reject({
-					isError: true,
-					message: "Error at deleteBranch(): " + e.message
-				});
 			});
-		});
+			return options.branchName + " has been deleted successfully.";
+		} catch(e) {
+			throw {
+				isError: true,
+				message: "Error at deleteBranch(): " + e.message
+			};
+		}
 	},
 
 	/**
 	 * An object that contains required information for fetching a file.
-	 * @typedef {Object} FetchRemoteFileOptions
+	 * @typedef {Object} FileOptions
 	 * @param {String} repoOwner - The repo owner.
 	 * @param {String} repoName - The repo name.
 	 * @param {String} branchName - The name of the remote branch to operate.
```

Comment:
I discovered through experiment that this member is optional - if it is omitted, the fetch will use the repository's default branch. We should document this member as optional